src/values/ECMAScriptFunctionValue.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Realm } from "../realm.js";
+import type { ObjectValue } from "./index.js";
+import { FunctionValue } from "./index.js";
+
+/* Abstract base class for non-exotic function objects(either with source or built-in) */
+export default class ECMAScriptFunctionValue extends FunctionValue {
+  constructor(realm: Realm, intrinsicName?: string) {
+    super(realm, intrinsicName);
+  }
+
+  $ConstructorKind: void | string;
```

Comment:
The spec says that this field is _Either "base" or "derived"._ Undefined is not an option.
Can you figure out the built-in functions should be?